contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -32,22 +32,29 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
 
         bytes32 checkpointHash = keccak256(abi.encode(checkpoint));
 
-        if (checkpoint.blockHeight == s.lastBottomUpCheckpointHeight + s.bottomUpCheckPeriod) {
```

Comment:
This will not work for a full msg batch. Because a message batch is created if the max number of bottom up cross messages are received, and a new checkpoint will be created, so the checkpoint epoch will not be at the checkpoint period.